apis/python/src/tiledbsoma/_sparse_nd_array.py

```diff
@@ -411,24 +431,173 @@ def coos(self, shape: Optional[NTuple] = None) -> SparseCOOTensorReadIter:
         """
         if shape is not None and (len(shape) != len(self.shape)):
             raise ValueError(f"shape must be a tuple of size {len(self.shape)}")
+        self.array._set_reader_coords(self.sr, self.coords)
         return SparseCOOTensorReadIter(self.sr, shape or self.shape)
 
-    def dense_tensors(self) -> somacore.ReadIter[pa.Tensor]:
+    def tables(self) -> TableReadIter:
         """
-        Returns an iterator of `Arrow Tensor <https://arrow.apache.org/docs/cpp/api/tensor.html>`_.
+        Returns an iterator of
+        `Arrow Table <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`_.
 
         Lifecycle:
             Experimental.
         """
+        self.array._set_reader_coords(self.sr, self.coords)
+        return TableReadIter(self.sr)
+
+    def blockwise(
+        self,
+        axis: Union[int, Sequence[int]],
+        *,
+        size: Optional[Union[int, Sequence[int]]] = None,
+        reindex_disable: Optional[Union[int, Sequence[int]]] = None,
+        eager: bool = True,
+    ) -> SparseNDArrayBlockwiseRead:
+        """
+        Returns an intermediate type to choose a blockwise iterator of a specific format.
 
-        raise NotImplementedError()
+        Blockwise iterators yield results grouped by a user-specified axis. For example, a
+        blockwise iterator with `axis=0` will yield results containing all coordinates for
+        a given "row" in the array, regardless of the read `result_order` (i.e., the sort order).
 
-    def tables(self) -> TableReadIter:
+        Blockwise iterators yield an array "block" in some user-specified format, as well as a
+        list of coordinates contained in the individual block.
+
+        All blockwise iterators will reindex coordinates (i.e., map them from soma_joinid to an integer
+        in the range [0, N)), unless reindexing is specifically disabled for that axis, using the
+        `reindex_disable` argument. When reindexing:
+        * the primary iterator axis coordinates, as indicated by the `axis` argument, will be reindexed into the range
+          `[0, N)`, where `N` is the number of coordinates read for the block (controlled with the `size` argument).
+        * all other axes will be reindexed will be reindexed to `[0, N)`, where `N` is the number of points read
+          on that axis across all blocks.
+
+        Args:
+            axis:
+                Required. The axis across which to yield blocks, indicated as the dimension number, e.g.,
+                `axis=0` will step across `soma_dim_0` (the first dimension).
+            size:
+                Optional. Number of coordinates in each block yielded by the iterator.
```

Comment:
I had to drill down a few levels to see what the default is. Worth documenting here?

Reply:
as it is a heuristic, I was reticent to add a hard-coded default.  But I can certainly weasel-word that...

Reply:
added, PTAL.

Reply:
Looks good, ty!  

(It's too bad SOMA doesn't have a central list of config settings like [TileDB config settings](https://docs.tiledb.com/main/how-to/configuration#configuration-parameters), where default values can be published to docs from code-level constants.)